### PR TITLE
Issues #113, #242 and #243: Arc modes (OPEN, CHORD, PIE)

### PR DESCRIPTION
**Pull request by atgeflu**
Hi guys,
Taking into consideration some of the comments from the [pull request for issue #113](https://github.com/processing-js/processing-js/pull/120) I implemented OPEN, CHORD and PIE modes for arc(). This should fix issues #113, #242 and #243.
Since I couldn't think of a way to unit test the changes I've created a few reference tests instead (the 3 OPEN, CHORD and PIE examples from the processing reference for arc() as well as a PIE example with negative start degree).

All the arc examples from the original processing reference for arc() work now.

Please let me know if there is something else I can do to improve the pull request.

Thanks,
Cheers
Jean-Noël

### CODE REVIEW

**Comment by Pomax** [comment]
Looks pretty good, can you also amend the `./test/tests.js` file to add this into it (near some other arc tests):
```
  { path: "arc-chord.pde",  tags: ["2D", "endShape2D", "Crisp"] },
  { path: "arc-open.pde",  tags: ["2D", "endShape2D", "Crisp"] },
  { path: "arc-pie.pde",  tags: ["2D", "endShape2D", "Crisp"] },
  { path: "arc-pie-negative-angle.pde",  tags: ["2D", "endShape2D", "Crisp"] },
```
This adds the test cases to the list of cases to run. They'll fail, because of how Pjs isn't drawing super crisp geometry, but that's fine: ref tests are for human evaluation.

**Comment by atgeflu** [comment]
Thanks a lot for having a look at my PR. 
I really appreciate you taking the time to give me feedback during your busy work week.
 
I added the test cases to /test/ref/tests.js near some other arc tests (I assume that's the right file since it seemed to be the only one containing lines like { path: ...).

When running node test now the 4 new ref tests are now visible in the browser. 
On my machine the tests passed though. Is it because I generated the ref tests in my browser and thus have ref images that are not particularly crisp to start with?

File: processing.js [hunk 1]
```diff
@@ -905,9 +911,9 @@ module.exports = function finalizeProcessing(Processing, options) {
    * source and bind to canvas via new Processing(canvas, sourcestring).
    * @param {CANVAS} canvas The html canvas element to bind to
    * @param {String[]} source The array of files that must be loaded
-   * @param {Function} complete A callback, called with the sketch as the argument.
+   * @param {Function} onComplete A callback, called with the sketch as the argument.
```
Comment (Pomax) [inline processing.js]: let's rename this back to `complete`, to keep with the "one issue, one fix" philosophy: this is a change technically not related to getting chord/pie closing to work, so doesn't really belong in this PR

**Comment by Pomax** [comment]
different combinations of browser and graphics cards behave slightly differently (nvidia cards vs amd cards in particular) so it's possible that your graphics system renders it closer to the intended output than mine did. 

If you can undo the `onComplete` rename (which is technically not related to fixing this issue) I'll be happy to merge in your changes and generate a new minor version of the library

**Comment by atgeflu** [comment]
For some reason I can't yet make out I get things like `onComplete` when I build processing.js with `browserify build.js -o processing.js`. Running `node test` fails to build processing.js on my system, which is why I used the manual operation instead. 
I'll try to figure out how to build processing.js without it introducing renames or I'll change it back manually.

**Comment by atgeflu** [comment]
I got rid of the changes manually for now. That way only the changes relevant to the fix are now in processing.js.
I'll have to see if I can figure out why it does these things in the first place.

**Comment by atgeflu** [comment]
I think I figured out the reason why processing.js has now `onComplete` instead of `complete`.
There was a change to [src/Helpers/finalizeProcessing.js](https://github.com/processing-js/processing-js/commit/574649d8c61980429dbb1f1c66feeea460078102) on the 19th Sept 2015 that renamed `complete` to `onComplete`. The last version of processing.js is from the 18th of Sept 2015 that's why it doesn't contain `onComplete`.
Maybe the best thing would be to merge my changes as they are and then generate processing.js anew.

**Comment by Pomax** [comment]
interesting. If this is a change that happened outside your own code, then we can leave it in, but then it's quite curious that the code on `master` does not show this as already being in the current codebase.

**Comment by atgeflu** [comment]
Considering that the fix has been merged now, would it make sense to close the issues #113, #242 and #243? 
Furthermore the PR #120 is now no longer needed, should it also be closed perhaps?

**Comment by Pomax** [comment]
well spotted: closed with notes on why.
